amp-request-signer/src/main/java/org/apache/flink/connector/prometheus/sink/aws/AWS4SignerBase.java

```diff
@@ -0,0 +1,290 @@
+     * @param httpMethod The HTTP verb for the request, e.g. GET.
+     * @param serviceName The signing name of the service, e.g. 's3'.
+     * @param regionName The system name of the AWS region associated with the endpoint, e.g.
+     *     us-east-1.
+     */
+    public AWS4SignerBase(
+            URL endpointUrl, String httpMethod, String serviceName, String regionName) {
+        this.endpointUrl = endpointUrl;
+        this.httpMethod = httpMethod;
+        this.serviceName = serviceName;
+        this.regionName = regionName;
+
+        dateTimeFormat = new SimpleDateFormat(ISO_8601_BASIC_FORMAT);
+        dateTimeFormat.setTimeZone(new SimpleTimeZone(0, "UTC"));
+        dateStampFormat = new SimpleDateFormat(DATE_STRING_FORMAT);
+        dateStampFormat.setTimeZone(new SimpleTimeZone(0, "UTC"));
+    }
+
+    /**
+     * Returns the canonical collection of header names that will be included in the signature. For
+     * AWS4, all header names must be included in the process in sorted canonicalized order.
+     */
+    protected static String getCanonicalizeHeaderNames(Map<String, String> headers) {
+        List<String> sortedHeaders = new ArrayList<String>();
+        sortedHeaders.addAll(headers.keySet());
+        Collections.sort(sortedHeaders, String.CASE_INSENSITIVE_ORDER);
+
+        StringBuilder buffer = new StringBuilder();
+        for (String header : sortedHeaders) {
+            if (buffer.length() > 0) {
+                buffer.append(";");
+            }
+            buffer.append(header.toLowerCase());
+        }
+
+        return buffer.toString();
+    }
+
+    /**
+     * Computes the canonical headers with values for the request. For AWS4, all headers must be
+     * included in the signing process.
+     */
+    protected static String getCanonicalizedHeaderString(Map<String, String> headers) {
+        if (headers == null || headers.isEmpty()) {
+            return "";
+        }
+
+        // step1: sort the headers by case-insensitive order
+        List<String> sortedHeaders = new ArrayList<String>();
+        sortedHeaders.addAll(headers.keySet());
+        Collections.sort(sortedHeaders, String.CASE_INSENSITIVE_ORDER);
+
+        // step2: form the canonical header:value entries in sorted order.
+        // Multiple white spaces in the values should be compressed to a single
+        // space.
+        StringBuilder buffer = new StringBuilder();
+        for (String key : sortedHeaders) {
+            buffer.append(
+                    key.toLowerCase().replaceAll("\\s+", " ")
+                            + ":"
+                            + headers.get(key).replaceAll("\\s+", " "));
+            buffer.append("\n");
+        }
+
+        return buffer.toString();
+    }
+
+    /**
+     * Returns the canonical request string to go into the signer process; this consists of several
+     * canonical sub-parts.
+     *
+     * @return
```

Comment:
Removed `@return`, that was actually redundant with the description above

Reply:
@z3d1k not sure I see the value of generating the Protobuf classes from `.proto` file instead of hardwiring the generated classes, in this context. 
The remote-write protocol is much more than the payload schema.
Any Prometheus remote-write protocol change will probably require changes to the code to adjust the behavior. 
Also, generating the code from `.proto` would require to do the same also with GoGoProto.
This is the reason all Java Prometheus remote-write clients I found embed the generated classes in the code. e.g. https://github.com/m3dbx/prometheus_remote_client_java but you can find others

Reply:
Discussed offline, my comment was about including _both_ generated version and source `*.proto` with a way to regenerate code.